### ISSUE TITLE
MacOS Catalyst Support

### ISSUE THREAD

**Issue by ajnozari** [react-native-webrtc/react-native-webrtc#1033]
#### Expected behavior
Building for macOS Big Sur or higher should be possible.

#### Observerd behavior
XCFramework is not detected and loaded for catalyst.  This does not affect iOS simulators

#### Steps to reproduce the problem
install the package and attempt to build with the target set to macOS.

#### Platform information
The xcframework needs to be told to link them for macOS.  In fact it should be able to use the same framework used for the simulator.  Ninja can even do this, but I'm not sure how to add macOS as a target to the xcFramework build script that is used.  I've tried several times to get it to build but I'm just not familiar enough with Ninja and how your build script is setup.

* **React Native version**: 0.64.2
* **Plugin version**: 0.89.3
* **OS**: MacOS
* **OS version**: 10.15+

**Comment by saghul** [comment]
PRs are welcome, it’s unlikely I’ll have the time to work on this myself.

**Comment by ajnozari** [comment]
So I was able to get ninja to start generating the macos catalyst slice but it failed halfway.  

I’m gonna give the errors a poke over time and see how far along I can push the build, however it seems to be an OpenGL function call which is deprecated and won’t run on catalyst, which might end this attempt until upstream webrtc themselves remove any OpenGL code.

I’ll post the exact error later when I can get it to generate the build error again. I know it’s been mentioned in other issues that this webRTC does use OpenGL so I’m wondering if maybe that means ninja is trying to swap something out on the fly and it’s not using Metal like apple now expects.

**Comment by saghul** [comment]
I just saw that Signal has contributed patches for building for Catalyst to WebRTC and they landed in master already.

There is a mention about some problems on x64 but arm64 seems to work fine.

I hope this matures enough in the next 1-2 WebRTC releases so we can pick it up.

**Comment by Flam3rboy** [comment]
Any updates on this?

**Comment by saghul** [comment]
Not really, sorry, I didn't get the time to investigate.

**Comment by Flam3rboy** [comment]
If this doesn't work, I'll have to use a fallback option. 
@saghul Do you know if it works if I embed a react-native-WebView and use the system browser WebRTC capabilities?

**Comment by saghul** [comment]
I don't know. You'd have to ask the WebView maintainers.

**Comment by Flam3rboy** [comment]
still searching for solutions:
```
'WebRTC/RTCCameraVideoCapturer.h' file not found
```

**Comment by Flam3rboy** [comment]
fixed it by installing pods with a x86 shell:
```
arch -x86_64 /bin/bash
cd react-native-project/ios
pod install
```

**Comment by Flam3rboy** [comment]
still searching for solutions

**Comment by SamuelScheit** [comment]
Hi @saghul, thank you so much for all your hard work on this library.
I recently came across another [WebRTC Pod](https://github.com/stasel/WebRTC) that has support for macOS Catalyst. 
Would it be possible for you to replicate this feature in Jitsi's WebRTC?
This library is currently the only obstacle preventing us from building our React-Native app for macOS Catalyst.
We would greatly appreciate it if you could find a solution so we don't have to rely on a less optimal Electron/web build. Thank you!

**Comment by saghul** [comment]
There is a comment in the build script of that project saying it doesn't work properly yet. Is that still the case? Have you tested that Pod?

**Comment by SamuelScheit** [comment]
still searching for solutions

**Comment by 8BallBomBom** [comment]
@saghul As `react-native-macos` has started churning out new versions, currently sat at `0.71.2` any chance of a re-assessment to see if we can get the macOS build going again?

Was looking to test and update the macOS example.

**Comment by saghul** [comment]
This issue is about Mac Catalyst, so iOS apps on macOS not the native macOS build itself.

I tried with the latest build but it needs some tweaks for M1 support, and I didn't have time to dig into it...

**Comment by 8BallBomBom** [comment]
Figured we could possibly also tackle Mac Catalyst in-between sorting out macOS support.
But was mostly trying to figure out the exact statuses 👍🏻

**Comment by saghul** [comment]
It's another type of target entirely, but I think upstream was fixed so we can give it a try.

**Comment by 8BallBomBom** [comment]
True, was mainly trying to bring focus to the macOS platform but thought Catalyst might also be viable 🤔
I'm willing to test builds and update the current examples.
Then it will be on to developing some new examples 😉

**Comment by SamuelScheit** [comment]
It looks like the latest webrtc version is compatible with macos catalyst.
By adding modifying `tools/build-webrtc.py`:
- add `catalyst:arm64` to `IOS_BUILD_ARCHS`
- change `ios_deployment_target="12.0"` to `ios_deployment_target="14.0"` (TODO: this should be only changed for catalyst, ios should stay on 12.0)

@saghul as webrtc now builds for catalyst, can you make a new release with maccatalyst support?

**Comment by saghul** [comment]
On what version did it land? I'd take a PR with the necessary changes to the build script.

**Comment by SamuelScheit** [comment]
I'm not sure on what version exactly, but it works with the latest one.
I'll make a pr for the build script 👍

**Comment by saghul** [comment]
Latest as on their repo or latest we use so 111?
